### ISSUE TITLE
Consider preventing players in the same lobby from choosing the same name

### ISSUE THREAD

**Issue by abcdefg30** [OpenRA/OpenRA#17670]
We might be able to extend the existing player name sanitization for this.
When a player with an already existing name joins a game we could immediately rename them and add a character (like `_`) to their name.

**Comment by anjew175** [comment]
I know source games add a '(1)' to the player that joins after, like how windows does for duplicate files.
I think applying a suffix is a better than preventing the person from choosing the name as I could see it being used for further trolling. eg. joining a game before that actual person joins in order to prevent their name being valid

**Comment by dragunoff** [comment]
Adding a number in brackets seems like a good solution because it can handle any number of duplicates and people are familiar with it from other games (source-based at least) and from operating systems (Windows at least).

**Comment by pchote** [comment]
Two problems come to mind:
* What happens if the username is already at the maximum length limit
* This will be useless against anyone acting maliciously - there are many ways to trivially work around this (adding spaces/invisible characters, unicode lookalike characters, etc) that we will not be able to guard against.

**Comment by abcdefg30** [comment]
The primary idea here was to make it easier to see who is who (who said what). It gets quite confusing when three people with the same name chat.

**Comment by pchote** [comment]
Yeah, this seems like a reasonable idea to cover the benign/accidental use cases.

**Comment by WhoCaresora** [comment]
I would recommend a way for auth people to have priority on their own name in that case. whether it is registering a "in game" nick in the forum settings or just same a the auth name.

Means I'm auth at rob (too soon ?), if there is already a rob in the server, he get renamed and not me because it maches my auth name (or the ingame nick linked to the auth).

**Comment by cool-pants** [comment]
1.For the same name problem what about color coding the names, like if two players with the same name appear we can probably change the color of their names in the chat or creating a small square of different colors near their name. We have about 20 easily distinguishable colors or else we can use symbols like star, circle,etc.
2. For the other character problem, i guess we can add character restrictions to visible characters for the player name.

PS: Please guide me as i am new to Git. Thank you.

**Comment by anjew175** [comment]
> Means I'm auth at rob (too soon ?), if there is already a rob in the server, he get renamed and not me because it maches my auth name (or the ingame nick linked to the auth).

Personally not a fan of giving someone priority because they registered a name first on the forums. You could easily circumvent the protection against duplicate forum accounts and poach the name before a player makes an account in order to troll them in game. I know at least 1 account named after a prolific player that is not owned by that person.

**Comment by KOYK** [comment]
> Two problems come to mind:
> 
> * What happens if the username is already at the maximum length limit

1) Reserve the last char from the maximum length limit, for adding a number.
2) Add predefined color to the said above^ last char.
